### PR TITLE
Upload with publisher: e-mail notifications to admins + test

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/publisher/backend.dart [hunk 1]
```diff
@@ -217,6 +217,20 @@ class PublisherBackend {
     });
   }
 
+  /// The list of e-mail addresses of the members with admin roles. The list
+  /// should be used to notify admins on upload events.
+  Future<List<String>> getAdminMemberEmails(String publisherId) async {
+    final result = <String>[];
+    final key = _db.emptyKey.append(Publisher, id: publisherId);
+    final query = _db.query<PublisherMember>(ancestorKey: key)
+      ..filter('role =', PublisherMemberRole.admin);
```
Comment (jonasfj) [inline app/lib/publisher/backend.dart]: Why not just email all members of the publisher... I don't think we need to filter on email...

File: app/lib/publisher/backend.dart [hunk 2]
```diff
@@ -217,6 +217,20 @@ class PublisherBackend {
...
+    await for (final m in query.run()) {
```
Comment (jonasfj) [inline app/lib/publisher/backend.dart]: Should we tweak `getEmailOfUserId` to accept a list of userIds and return a list of emails..
Wouldn't that be more efficient in terms of datastore operations?

File: app/test/package/backend_test.dart [hunk 3]
```diff
@@ -480,6 +482,56 @@ void main() {
           // TODO: check history
           // TODO: check assets
         });
+
+        testWithServices('package under publisher', () async {
```
Comment (jonasfj) [inline app/test/package/backend_test.dart]: Couldn't we cover this by hitting the API? thus, creating a test that we don't need to update when we migrate away from `pub_server`?

**Comment by jonasfj** [comment]
All comments, are nit, we can do them later --- or not at all, just thoughts :)

File: app/test/package/backend_test.dart (hunk 3 above)
Comment (isoos) [reply app/test/package/backend_test.dart]: We could, but it is much more work to get all the required boilerplate setup and going. `pkg/pub_integration` tests do that, not sure if we really need to have that kind of setup for all the tests.

File: app/lib/publisher/backend.dart (hunk 2 above)
Comment (isoos) [reply app/lib/publisher/backend.dart]: Good point, we already have a batching for it, switched to `getEmailsOfUserIds`. Under the hood it uses a cache too.

File: app/lib/publisher/backend.dart (hunk 1 above)
Comment (isoos) [reply app/lib/publisher/backend.dart]: Sure, let's start with that.

**Pull request merged**